libsession/src/main/java/org/session/libsession/snode/SnodeAPI.kt

```diff
@@ -173,12 +183,22 @@ object SnodeAPI {
                     if (rawSnodes != null) {
                         val snodePool = rawSnodes.mapNotNull { rawSnode ->
                             val rawSnodeAsJSON = rawSnode as? Map<*, *>
-                            val address = rawSnodeAsJSON?.get("public_ip") as? String
-                            val port = rawSnodeAsJSON?.get("storage_port") as? Int
-                            val ed25519Key = rawSnodeAsJSON?.get("pubkey_ed25519") as? String
-                            val x25519Key = rawSnodeAsJSON?.get("pubkey_x25519") as? String
-                            if (address != null && port != null && ed25519Key != null && x25519Key != null && address != "0.0.0.0") {
-                                Snode("https://$address", port, Snode.KeySet(ed25519Key, x25519Key))
+                            val address = rawSnodeAsJSON?.get(KEY_IP) as? String
+                            val port = rawSnodeAsJSON?.get(KEY_PORT) as? Int
+                            val ed25519Key = rawSnodeAsJSON?.get(KEY_ED25519) as? String
+                            val x25519Key = rawSnodeAsJSON?.get(KEY_X25519) as? String
+                            val version = (rawSnodeAsJSON?.get(KEY_VERSION) as? ArrayList<*>)
+                                ?.filterIsInstance<Int>() // get the array as Integers
+                                ?.joinToString(separator = ".") // turn it int a version string
+
+                            if (address != null && port != null && ed25519Key != null && x25519Key != null
+                                && address != "0.0.0.0" && version != null) {
```

Comment:
Could move address != "0.0.0.0" closer to the null check for it if you wanted, or not.